Controller/TreeController.php

```diff
@@ -85,21 +63,11 @@ public function __construct(
      */
     public function treeAction(Request $request)
     {
-        $createInOverlay = $request->attributes->get('create_in_overlay');
-        $editInOverlay = $request->attributes->get('edit_in_overlay');
-        $deleteInOverlay = $request->attributes->get('delete_in_overlay');
-
         $root = $request->attributes->get('root');
-        $selected = $request->attributes->get('selected') ?: $root;
 
         return $this->render($this->template, array(
-            'repository_name' => $this->repositoryName,
             'root_node' => $root,
-            'routing_defaults' => $this->defaults,
-            //'confirm_move' => $this->confirmMove,
-            //'create_in_overlay' => $createInOverlay ? $createInOverlay : false,
-            //'edit_in_overlay' => $editInOverlay ? $editInOverlay : false,
-            //'delete_in_overlay' => $deleteInOverlay ? $deleteInOverlay : false,
+            'tree_configuration' => $this->treeConfiguration,
```

Comment:
lets explode the array here and pass explicit named parameters to the template as before. then its clear what we expect to exist and the template will be simpler.